### DISCUSSION TITLE
Question: How can someone use this for transaction emails?

### DISCUSSION THREAD

**Discussion by patchthecode** [docker-mailserver/docker-mailserver#3616] [Q&A] [answered]
### Subject

I would like some feedback concerning a use case

### Description

This is a question (not a bug).
My app's user click on "Forgot Password".

I would like my app to talk to docker-mailserver to send an email to said user.  Does docker mail server have a solution for this? Or can it use some other project for this? I was thinking of using `Mautic`

**Comment by polarathene** [comment, marked as answer]
> I would like my app to talk to docker-mailserver to send an email to said user.

- Submit mail to submission ports (465 or 587) with an authenticated user (`setup email add ...`).
- The authenticated username does not have anything to do with the sender address (_except if `SPOOF_PROTECTION=1` is enabled_). So you can use whatever sender address you like (eg: `noreply@example.com`).
- App does the same thing, use email client/library to create an email and send to a submission port with the required credentials for sending.

After you handle that, the mail will be sent outbound on port 25 to be delivered to the recipient address (destination).

Usually to avoid issues with sender reputation you'll relay your mail through a service like SendGrid.
- In that case you configure DMS to relay outbound mail through the smarthost/relayhost, which may send mail to service like SendGrid on a different port (_non-standard mail ports they offer, or standard submission ports_).
- You could also avoid DMS entirely, and just send mail via their API, or submitting the mail to SendGrid. If you don't intend for DMS to receive mail to store and retrieve, then a send-only DMS is referred to as a null-mailer.

If you don't use services like SendGrid, your mail may not be delivered or more likely to arrive in spam/junk for the user. Be advised that many VPS prevent sending mail out on port 25, so you may need to rely on a relay service regardless.

---

**TL;DR:**
- DMS provides no API.
- DMS can send the mail, but you must submit it as mail to DMS through a mail client / library.

I am moving this to Q&A discussion, as it's a question.

**Comment by patchthecode** [comment]
So if i understand correctly, this should be a client's job.
I have to get a mail client (that works with DMS). And the client will be the one to send message to dms, which will then send the email?

If yes, then thanks.
I just have one final question, I have searched this repository to see if `Mautic` client can be configured to send email with DMS (just like rooundcube does), but i found nothing. Do you know the location of any such documentation? If no, then its ok, I will keep searching.

**Reply by polarathene** [reply]
> I have to get a mail client (that works with DMS). And the client will be the one to send message to dms, which will then send the email?

Yes, just like if you send mail with ThunderBird, you can use a library in your preferred language that lets you use an API to send mail in the same way.

The mail is then submitted through DMS to be sent to another mail-server. DMS will handle the rest on behalf of the mail client.

>Do you know the location of any such documentation? If no, then its ok, I will keep searching.

I am not sure where they document the configuration, but [they mention you can use the email services](https://docs.mautic.org/en/channels/emails#email-delivery) like I mentioned earlier (SendGrid), or any other mail server (DMS).

Usually you should be able to configure the mail client to connect to DMS with username and password, just like you would with software like Thunderbird.

It might be relying on something from the system to handle this instead, I tried a few quick searches on their docs but nothing came up for me. Perhaps it's GUI only settings? Try contacting Mautic for support on that.

You shouldn't have to worry about compatibility with DMS, this is standard SMTP server (MTA) functionality.
